<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> 9 | Functional Gene Annotation with METABOLIC | Osburn Lab Protocols</title>
  <meta name="description" content="This is a collection of protocols for the Osburn Lab." />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content=" 9 | Functional Gene Annotation with METABOLIC | Osburn Lab Protocols" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a collection of protocols for the Osburn Lab." />
  <meta name="github-repo" content="rstudio/osburnlab/protocols" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" 9 | Functional Gene Annotation with METABOLIC | Osburn Lab Protocols" />
  
  <meta name="twitter:description" content="This is a collection of protocols for the Osburn Lab." />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="submit-sequence-data.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
<link rel="stylesheet" href="font-awesome.min.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li>

<a href="./">
<img class="logo" src="images/LabLogo_White-01.png" height="50">
</a>
</li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> | About</a></li>
<li class="chapter" data-level="2" data-path="data-access.html"><a href="data-access.html"><i class="fa fa-check"></i><b>2</b> | Data Access + Storage</a><ul>
<li class="chapter" data-level="2.1" data-path="data-access.html"><a href="data-access.html#macos-users"><i class="fa fa-check"></i><b>2.1</b> MacOS Users</a></li>
<li class="chapter" data-level="2.2" data-path="data-access.html"><a href="data-access.html#windows-users"><i class="fa fa-check"></i><b>2.2</b> Windows Users</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="version-control.html"><a href="version-control.html"><i class="fa fa-check"></i><b>3</b> | Version Control</a></li>
<li class="chapter" data-level="4" data-path="create-a-protocol.html"><a href="create-a-protocol.html"><i class="fa fa-check"></i><b>4</b> | Create a Protocol</a></li>
<li class="chapter" data-level="5" data-path="qiime2-workflow.html"><a href="qiime2-workflow.html"><i class="fa fa-check"></i><b>5</b> | Qiime2 workflow</a><ul>
<li class="chapter" data-level="5.1" data-path="qiime2-workflow.html"><a href="qiime2-workflow.html#import-data"><i class="fa fa-check"></i><b>5.1</b> Import data</a></li>
<li class="chapter" data-level="5.2" data-path="qiime2-workflow.html"><a href="qiime2-workflow.html#demultiplexing"><i class="fa fa-check"></i><b>5.2</b> Demultiplexing</a></li>
<li class="chapter" data-level="5.3" data-path="qiime2-workflow.html"><a href="qiime2-workflow.html#denoising-and-asv-generation"><i class="fa fa-check"></i><b>5.3</b> Denoising and ASV generation</a></li>
<li class="chapter" data-level="5.4" data-path="qiime2-workflow.html"><a href="qiime2-workflow.html#taxonomy"><i class="fa fa-check"></i><b>5.4</b> Taxonomy</a></li>
<li class="chapter" data-level="5.5" data-path="qiime2-workflow.html"><a href="qiime2-workflow.html#taxa-barplots-and-diversity-analyses-in-qiime2"><i class="fa fa-check"></i><b>5.5</b> Taxa barplots and diversity analyses in Qiime2</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="quest-tutorial.html"><a href="quest-tutorial.html"><i class="fa fa-check"></i><b>6</b> | Quest tutorial</a><ul>
<li class="chapter" data-level="6.1" data-path="quest-tutorial.html"><a href="quest-tutorial.html#getting-acquainted-with-quest"><i class="fa fa-check"></i><b>6.1</b> Getting acquainted with Quest</a></li>
<li class="chapter" data-level="6.2" data-path="quest-tutorial.html"><a href="quest-tutorial.html#using-qiime2-on-quest"><i class="fa fa-check"></i><b>6.2</b> Using Qiime2 on Quest</a></li>
<li class="chapter" data-level="6.3" data-path="quest-tutorial.html"><a href="quest-tutorial.html#best-practices-in-a-shared-computing-environment"><i class="fa fa-check"></i><b>6.3</b> Best practices in a shared computing environment</a></li>
<li class="chapter" data-level="6.4" data-path="quest-tutorial.html"><a href="quest-tutorial.html#interactive-jobs-on-quest"><i class="fa fa-check"></i><b>6.4</b> Interactive jobs on Quest</a></li>
<li class="chapter" data-level="6.5" data-path="quest-tutorial.html"><a href="quest-tutorial.html#batch-jobs-on-quest"><i class="fa fa-check"></i><b>6.5</b> Batch jobs on Quest</a></li>
<li class="chapter" data-level="6.6" data-path="quest-tutorial.html"><a href="quest-tutorial.html#a-note-on-partitions"><i class="fa fa-check"></i><b>6.6</b> A note on partitions</a></li>
<li class="chapter" data-level="6.7" data-path="quest-tutorial.html"><a href="quest-tutorial.html#more-information"><i class="fa fa-check"></i><b>6.7</b> More information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="DAPI.html"><a href="DAPI.html"><i class="fa fa-check"></i><b>7</b> | DAPI + Cell Counting</a><ul>
<li class="chapter" data-level="7.1" data-path="DAPI.html"><a href="DAPI.html#sample-prep"><i class="fa fa-check"></i><b>7.1</b> Sample prep</a></li>
<li class="chapter" data-level="7.2" data-path="DAPI.html"><a href="DAPI.html#operating-the-microscope"><i class="fa fa-check"></i><b>7.2</b> Operating the Microscope</a></li>
<li class="chapter" data-level="7.3" data-path="DAPI.html"><a href="DAPI.html#cell-counting"><i class="fa fa-check"></i><b>7.3</b> Cell Counting</a></li>
<li class="chapter" data-level="7.4" data-path="DAPI.html"><a href="DAPI.html#troubleshooting"><i class="fa fa-check"></i><b>7.4</b> Troubleshooting</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html"><i class="fa fa-check"></i><b>8</b> | Submit Sequence Data</a><ul>
<li class="chapter" data-level="8.1" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#new-submission"><i class="fa fa-check"></i><b>8.1</b> New Submission</a></li>
<li class="chapter" data-level="8.2" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#subission-type"><i class="fa fa-check"></i><b>8.2</b> 1 Subission Type</a></li>
<li class="chapter" data-level="8.3" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#submitter"><i class="fa fa-check"></i><b>8.3</b> 2 Submitter</a></li>
<li class="chapter" data-level="8.4" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#sequencing-technology"><i class="fa fa-check"></i><b>8.4</b> 3 Sequencing Technology</a></li>
<li class="chapter" data-level="8.5" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#sequences"><i class="fa fa-check"></i><b>8.5</b> 4 Sequences</a></li>
<li class="chapter" data-level="8.6" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#sequence-processing"><i class="fa fa-check"></i><b>8.6</b> 5 Sequence Processing</a></li>
<li class="chapter" data-level="8.7" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#source-info"><i class="fa fa-check"></i><b>8.7</b> 6 Source Info</a></li>
<li class="chapter" data-level="8.8" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#bioproject-info"><i class="fa fa-check"></i><b>8.8</b> 7 BioProject Info</a></li>
<li class="chapter" data-level="8.9" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#biosample-type"><i class="fa fa-check"></i><b>8.9</b> 8 BioSample Type</a></li>
<li class="chapter" data-level="8.10" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#biosample-attributes"><i class="fa fa-check"></i><b>8.10</b> 9 BioSample Attributes</a></li>
<li class="chapter" data-level="8.11" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#references"><i class="fa fa-check"></i><b>8.11</b> 10 References</a></li>
<li class="chapter" data-level="8.12" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#review-submit"><i class="fa fa-check"></i><b>8.12</b> 11 Review &amp; Submit</a></li>
<li class="chapter" data-level="8.13" data-path="submit-sequence-data.html"><a href="submit-sequence-data.html#correcting-submission"><i class="fa fa-check"></i><b>8.13</b> Correcting Submission</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="metabolic.html"><a href="metabolic.html"><i class="fa fa-check"></i><b>9</b> | Functional Gene Annotation with METABOLIC</a></li>
<li class="divider"></li>
<li class="social">
<a target="blank" href="https://github.com/osburnlab" class="icon fa-github"></a>
<a target="blank" href="https://twitter.com/osburnlab" class="icon fa-twitter"></a>
</li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Osburn Lab Protocols</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="metabolic" class="section level1">
<h1><span class="header-section-number"> 9</span> | Functional Gene Annotation with METABOLIC</h1>
<p><font size="1"><strong>Created by:</strong> Caitlin Casar on 2021-01-20 </br>
<strong>Last updated:</strong> 2021-01-20 </font></p>
<p>This is a protocol for annotating your metagenome or binned genome data with <a href="https://github.com/AnantharamanLab/METABOLIC" target="_blank">METABOLIC</a>.</p>
<p>First you will need to install METABOLIC on Quest. Save the following to a .sh file to your home directory and run as a batch job on Quest:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb30-1" title="1"><span class="ex">module</span> load anaconda3</a>
<a class="sourceLine" id="cb30-2" title="2"></a>
<a class="sourceLine" id="cb30-3" title="3"><span class="co"># create env</span></a>
<a class="sourceLine" id="cb30-4" title="4"><span class="ex">conda</span> create -n metabolic</a>
<a class="sourceLine" id="cb30-5" title="5"><span class="bu">source</span> activate metabolic</a>
<a class="sourceLine" id="cb30-6" title="6"></a>
<a class="sourceLine" id="cb30-7" title="7"><span class="co"># conda install the required tools</span></a>
<a class="sourceLine" id="cb30-8" title="8"><span class="ex">conda</span> install sambamba</a>
<a class="sourceLine" id="cb30-9" title="9"><span class="ex">conda</span> install bamtools</a>
<a class="sourceLine" id="cb30-10" title="10"><span class="ex">conda</span> install coverm <span class="co"># installs perl 5.32</span></a>
<a class="sourceLine" id="cb30-11" title="11"><span class="ex">conda</span> install gtdbtk</a>
<a class="sourceLine" id="cb30-12" title="12"><span class="ex">conda</span> install diamond</a>
<a class="sourceLine" id="cb30-13" title="13"><span class="ex">conda</span> install bowtie2</a>
<a class="sourceLine" id="cb30-14" title="14"><span class="ex">conda</span> install R=3.6.0</a>
<a class="sourceLine" id="cb30-15" title="15"></a>
<a class="sourceLine" id="cb30-16" title="16"><span class="co"># conda install R dependencies</span></a>
<a class="sourceLine" id="cb30-17" title="17"><span class="ex">conda</span> install r-tidyverse=1.3.0</a>
<a class="sourceLine" id="cb30-18" title="18"><span class="ex">conda</span> install r-diagram</a>
<a class="sourceLine" id="cb30-19" title="19"><span class="ex">conda</span> install r-ggthemes</a>
<a class="sourceLine" id="cb30-20" title="20"><span class="ex">conda</span> install r-ggalluvial</a>
<a class="sourceLine" id="cb30-21" title="21"><span class="ex">conda</span> install r-ggraph</a>
<a class="sourceLine" id="cb30-22" title="22"><span class="ex">conda</span> install r-openxlsx</a>
<a class="sourceLine" id="cb30-23" title="23"><span class="ex">conda</span> install r-pdftools</a>
<a class="sourceLine" id="cb30-24" title="24"></a>
<a class="sourceLine" id="cb30-25" title="25"><span class="co"># conda install perl dependencies</span></a>
<a class="sourceLine" id="cb30-26" title="26"><span class="ex">conda</span> install perl-data-dumper <span class="co"># downgrades perl to 5.26.2</span></a>
<a class="sourceLine" id="cb30-27" title="27"><span class="ex">conda</span> install perl-excel-writer-xlsx</a>
<a class="sourceLine" id="cb30-28" title="28"><span class="ex">conda</span> install perl-posix</a>
<a class="sourceLine" id="cb30-29" title="29"><span class="ex">conda</span> install perl-getopt-long</a>
<a class="sourceLine" id="cb30-30" title="30"><span class="ex">conda</span> install perl-statistics-descriptive</a>
<a class="sourceLine" id="cb30-31" title="31"><span class="ex">conda</span> install perl-bioperl</a>
<a class="sourceLine" id="cb30-32" title="32"></a>
<a class="sourceLine" id="cb30-33" title="33"><span class="co"># get the one pesky perl dependency not available through conda</span></a>
<a class="sourceLine" id="cb30-34" title="34"><span class="ex">conda</span> install perl-app-cpanminus</a>
<a class="sourceLine" id="cb30-35" title="35"><span class="fu">env</span> PERL5LIB=<span class="st">&quot;&quot;</span> PERL_LOCAL_LIB_ROOT=<span class="st">&quot;&quot;</span> PERL_MM_OPT=<span class="st">&quot;&quot;</span> PERL_MB_OPT=<span class="st">&quot;&quot;</span> cpanm Array::Split</a>
<a class="sourceLine" id="cb30-36" title="36"></a>
<a class="sourceLine" id="cb30-37" title="37"><span class="co">#conda install gdown</span></a>
<a class="sourceLine" id="cb30-38" title="38"><span class="ex">conda</span> install gdown</a>
<a class="sourceLine" id="cb30-39" title="39"></a>
<a class="sourceLine" id="cb30-40" title="40"><span class="co"># conda install the perl package to solve the first (and so far only) error</span></a>
<a class="sourceLine" id="cb30-41" title="41"><span class="ex">conda</span> install perl-parallel-forkmanager</a>
<a class="sourceLine" id="cb30-42" title="42"></a>
<a class="sourceLine" id="cb30-43" title="43"><span class="co">#clone github repo</span></a>
<a class="sourceLine" id="cb30-44" title="44"><span class="fu">git</span> clone https://github.com/AnantharamanLab/METABOLIC.git</a>
<a class="sourceLine" id="cb30-45" title="45"></a>
<a class="sourceLine" id="cb30-46" title="46"><span class="co">#run setup script - make sure you are working from your metabolic conda environment or this script will fail</span></a>
<a class="sourceLine" id="cb30-47" title="47"><span class="bu">cd</span> METABOLIC</a>
<a class="sourceLine" id="cb30-48" title="48"><span class="fu">sh</span> run_to_setup.sh</a>
<a class="sourceLine" id="cb30-49" title="49"></a>
<a class="sourceLine" id="cb30-50" title="50"><span class="co">#test installation - this is optional but good for troubleshooting installation errors</span></a>
<a class="sourceLine" id="cb30-51" title="51"><span class="fu">perl</span> /home/cpc7770/METABOLIC/METABOLIC-G.pl -test true</a>
<a class="sourceLine" id="cb30-52" title="52"><span class="fu">perl</span> /home/cpc7770/METABOLIC/METABOLIC-C.pl -test true</a></code></pre></div>
<p>Now you’re ready to run METABOLIC on your metagenome data. If you want to run it on unbinned metagenome data, follow this example batch script. Note that for <code>METABOLIC-G.pl</code> you only need to specify the input and output directory. You can run it on either nucleotide files with a <code>.fasta</code> extension, or amino acid files with a <code>.faa</code> extension.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb31-1" title="1"><span class="ex">module</span> load anaconda3</a>
<a class="sourceLine" id="cb31-2" title="2"><span class="bu">source</span> activate metabolic</a>
<a class="sourceLine" id="cb31-3" title="3"></a>
<a class="sourceLine" id="cb31-4" title="4"><span class="co">#run on metagenomes - note that the file extensions MUST be &#39;.fasta&#39;, &#39;.fa&#39; is not acceptable.</span></a>
<a class="sourceLine" id="cb31-5" title="5"><span class="fu">perl</span> /home/cpc7770/METABOLIC/METABOLIC-G.pl -in-gn /projects/p30777/metagenome_data/contigs -o /projects/p30777/metagenome_data/metabolic_metagenome_annotations</a></code></pre></div>
<p>Wanna run it on binned genomes? Use the <code>METABOLIC-C.pl</code> script. Note that for this script you need to specify the paired read files. You can run the following as a batch job on Quest:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb32-1" title="1"><span class="co">#run on binned genomes</span></a>
<a class="sourceLine" id="cb32-2" title="2"></a>
<a class="sourceLine" id="cb32-3" title="3"><span class="co">#rename from .fa to .fasta</span></a>
<a class="sourceLine" id="cb32-4" title="4"><span class="co">#cd /projects/p30777/metagenome_data/genomes</span></a>
<a class="sourceLine" id="cb32-5" title="5"><span class="co">#find . -name &quot;*.fa&quot; -exec rename .fa .fasta {} +</span></a>
<a class="sourceLine" id="cb32-6" title="6"></a>
<a class="sourceLine" id="cb32-7" title="7"></a>
<a class="sourceLine" id="cb32-8" title="8"><span class="co">#run on metagenomes - note that the file extensions MUST be &#39;.fasta&#39;, &#39;.fa&#39; is not acceptable.</span></a>
<a class="sourceLine" id="cb32-9" title="9"><span class="co">#loop over directories </span></a>
<a class="sourceLine" id="cb32-10" title="10"><span class="kw">for</span> <span class="ex">path</span> in /projects/p30777/metagenome_data/genomes/*<span class="kw">;</span> <span class="kw">do</span></a>
<a class="sourceLine" id="cb32-11" title="11">    [ <span class="ex">-d</span> <span class="st">&quot;</span><span class="va">${path}</span><span class="st">&quot;</span> ] <span class="kw">||</span> <span class="bu">continue</span> <span class="co"># if not a directory, skip</span></a>
<a class="sourceLine" id="cb32-12" title="12">    <span class="va">dirname=</span><span class="st">&quot;</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">&quot;</span><span class="va">${path}</span><span class="st">&quot;</span><span class="va">)</span><span class="st">&quot;</span></a>
<a class="sourceLine" id="cb32-13" title="13">    <span class="bu">echo</span> annotating <span class="va">$dirname</span> genomes...</a>
<a class="sourceLine" id="cb32-14" title="14">    <span class="fu">perl</span> ~/METABOLIC/METABOLIC-C.pl -in-gn <span class="va">$path</span>  -r /projects/p30777/metagenome_data/reads/<span class="va">$dirname</span> -o /projects/p30777/metagenome_data/metabolic_genome_annotations/<span class="va">$dirname</span></a>
<a class="sourceLine" id="cb32-15" title="15"><span class="kw">done</span></a>
<a class="sourceLine" id="cb32-16" title="16"></a>
<a class="sourceLine" id="cb32-17" title="17"><span class="bu">echo</span> done!</a></code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="submit-sequence-data.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/OsburnLab/Protocols/edit/master/08-metabolic.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
